plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/rest/TestTrinoRESTCatalogConnectorTest.java

```diff
@@ -0,0 +1,170 @@
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.http.server.testing.TestingHttpServer;
+import io.trino.plugin.iceberg.IcebergQueryRunner;
+import io.trino.plugin.iceberg.TestIcebergParquetConnectorTest;
+import io.trino.testing.QueryRunner;
+import io.trino.testing.TestingConnectorBehavior;
+import io.trino.tpch.TpchTable;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.rest.DelegatingRESTSessionCatalog;
+import org.assertj.core.util.Files;
+import org.testng.SkipException;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.util.Optional;
+
+import static io.trino.plugin.iceberg.IcebergFileFormat.PARQUET;
+import static io.trino.tpch.TpchTable.LINE_ITEM;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestTrinoRESTCatalogConnectorTest
+        extends TestIcebergParquetConnectorTest
+{
+    @Override
+    protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
+    {
+        switch (connectorBehavior) {
+            case SUPPORTS_RENAME_SCHEMA:
+            case SUPPORTS_CREATE_VIEW:
+            case SUPPORTS_COMMENT_ON_VIEW:
+            case SUPPORTS_CREATE_MATERIALIZED_VIEW:
+            case SUPPORTS_RENAME_MATERIALIZED_VIEW:
+                return false;
+            default:
+                return super.hasBehavior(connectorBehavior);
+        }
+    }
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        File warehouseLocation = Files.newTemporaryFolder();
+        warehouseLocation.deleteOnExit();
+
+        Catalog backend = RESTCatalogTestUtils.backendCatalog(warehouseLocation);
+
+        DelegatingRESTSessionCatalog delegatingCatalog = DelegatingRESTSessionCatalog
+                .builder().delegate(backend).build();
+
+        TestingHttpServer testServer = delegatingCatalog.testServer();
+        testServer.start();
+        closeAfterClass(testServer::stop);
+
+        return IcebergQueryRunner.builder()
+                .setBaseDataDir(Optional.of(warehouseLocation.toPath()))
+                .setIcebergProperties(
+                        ImmutableMap.<String, String>builder()
+                                .put("iceberg.file-format", "PARQUET")
+                                .put("iceberg.catalog.type", "rest")
+                                .put("iceberg.rest.uri", testServer.getBaseUrl().toString())
+                                .buildOrThrow())
+                .setInitialTables(ImmutableList.<TpchTable<?>>builder()
+                        .addAll(REQUIRED_TPCH_TABLES)
+                        .add(LINE_ITEM)
+                        .build())
+                .build();
+    }
+
+    @Override
+    @Test
+    public void testCreateSchemaWithLongName()
+    {
+        throw new SkipException("We're hitting max URI length before hitting the limit of the backing catalog");
```

Comment:
What's the maximum name length of REST catalog? Could you override `maxSchemaNameLength` instead of skip?

Reply:
I don't think this makes sense to test this with the REST catalog. We are hitting 414 URI Too Long before we'd be hitting whatever the limit of the backend catalog is

Reply:
Then, please test with the backend catalog limitation (e.g. JDBC). Just skipping doesn't make sense.

Reply:
It is worth mentioning that the behavior in this backing catalog is different from what you're implementing in https://github.com/trinodb/trino/pull/11772 and comes down to implementation details in Iceberg's [`JdbcCatalog`](https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java) itself. 

The important piece that we are trying to show is that the REST catalog is properly working with Trino. So in this particular case the backing catalog behaves different because not all things are mapped 1:1 to the DB (e.g. creating a namespace does not create an actual schema in the DB) and therefore not the same limits apply that one would expect.

It is also worth mentioning that we have to use an Iceberg catalog as a backing catalog and `JdbcCatalog` is the easiest to use.

Reply:
> The important piece that we are trying to show is that the REST catalog is properly working with Trino.

Just skipping tests doesn't show how REST catalog works. You can add custom test code in the `testCreateSchemaWithLongName` with some code comment.

Reply:
We added tests for length for schema and tables.